benchmarks/tracing_trace.rb

```diff
@@ -0,0 +1,48 @@
+# Used to quickly run benchmark under RSpec as part of the usual test suite, to validate it didn't bitrot
+VALIDATE_BENCHMARK_MODE = ENV['VALIDATE_BENCHMARK'] == 'true'
+
+return unless __FILE__ == $PROGRAM_NAME || VALIDATE_BENCHMARK_MODE
+
+require 'benchmark/ips'
+
+# Load the local version of the gem
+lib = File.expand_path('lib', __dir__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'ddtrace'
+
+class TracingTraceBenchmark
+  module FauxWriter
+    def write(trace)
+      # no-op
+    end
+
+    ::Datadog::Tracing::Writer.prepend(self)
+  end
+
+  def run_benchmark
+    Benchmark.ips do |x|
+      benchmark_time = VALIDATE_BENCHMARK_MODE ? { time: 0.001, warmup: 0 } : { time: 10.5, warmup: 2 }
```

Comment:
Minor: 10.5 seconds for the benchmark seems like an oddly-specific number? :rofl: